### PR TITLE
IDEX-3093 improve linked mode, use it for rename refactoring

### PR DESCRIPTION
**Pull request by evidolob**

### CODE REVIEW

**Comment by skabashnyuk** [comment]
ok for me

File: ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/texteditor/EmbeddedTextEditorPresenter.java [hunk 1]
```diff
@@ -794,7 +794,35 @@ public boolean isFocused() {
...
     @Override
     public boolean isAutoSaveEnabled() {
+        ReconcilerWithAutoSave autoSave = getAutoSave();
+        return autoSave != null && autoSave.isAutoSaveEnabled();
+    }
+
+    private ReconcilerWithAutoSave getAutoSave(){
         Reconciler reconciler = getConfiguration().getReconciler();
-         return reconciler != null && reconciler instanceof ReconcilerWithAutoSave;
+        if(reconciler == null){
```
Comment (evoevodin) [inline ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/texteditor/EmbeddedTextEditorPresenter.java]: You can simplify this method removing this <code>if</code> statement & block

File: ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/texteditor/EmbeddedTextEditorPresenter.java [hunk 2]
```diff
@@ -794,7 +794,35 @@ public boolean isFocused() {
...
+        if(reconciler instanceof ReconcilerWithAutoSave) {
+            return ((ReconcilerWithAutoSave)reconciler);
+        }
+        return null;
+    }
+
+    @Override
+    public void enableAutoSave() {
+        ReconcilerWithAutoSave autoSave = getAutoSave();
+        if(autoSave != null) {
```
Comment (evoevodin) [inline ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/texteditor/EmbeddedTextEditorPresenter.java]: This source file is not formatted, space required after <code>if</code>

File: ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/reconciler/ReconcilerWithAutoSave.java [hunk 3]
```diff
@@ -87,27 +90,41 @@ public void install(TextEditor editor) {
         this.editor = editor;
         this.dirtyRegionQueue = new DirtyRegionQueue();
         reconcilerDocumentChanged();
+        editor.addPropertyListener(new PropertyListener() {
```
Comment (evoevodin) [inline ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/reconciler/ReconcilerWithAutoSave.java]: Listener which does nothing, looks weird for me. Is it okay?

File: ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/link/LinkedModel.java [hunk 4]
```diff
@@ -14,6 +14,10 @@
 import java.util.List;
 
 /**
+ * The model for linked mode, umbrellas several
```
Comment (evoevodin) [inline ide/che-core-ide-jseditor/src/main/java/org/eclipse/che/ide/jseditor/client/link/LinkedModel.java]: This javadoc may be formatted in better way.

* Separate description from details.
* Separate <code>@author</code> from other stuff 

for instance:

```java
 /**
  * The model for linked mode, umbrellas several {@link LinkedModelGroup model groups}.
  * 
  * <p>Once installed ,the model propagates all changes 
  * to its siblings in the same position group.
  * 
  * @author Evgen Vidolob
  */
```

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/editor/EditorWithAutoSave.java [hunk 5]
```diff
@@ -11,9 +11,24 @@
...
 
 /**
+ * Editor auto save functionality.
+ * It's supports enable/disable auto save.
  * @author Evgen Vidolob
  */
 public interface EditorWithAutoSave {
 
+    /**
+     * @return true if auto save is enabled, false otherwise.
```
Comment (evoevodin) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/editor/EditorWithAutoSave.java]: See [The summary fragment -> Tip](https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment). Actually i am working on che code style spec and it is the same in this case.

**Pull request merged**